code/_globalvars/logging.dm

```diff
@@ -1,55 +1,73 @@
 GLOBAL_LIST_EMPTY(active_turfs_startlist)
 
-GLOBAL_LIST_EMPTY(admin_log)
-GLOBAL_PROTECT(admin_log)
-
-GLOBAL_LIST_EMPTY(adminlog)
-GLOBAL_PROTECT(adminlog)
-
-GLOBAL_LIST_EMPTY(bombers)
-GLOBAL_PROTECT(bombers)
-
-GLOBAL_LIST_EMPTY(combatlog)
-GLOBAL_PROTECT(combatlog)
-
-GLOBAL_VAR(config_error_log)
-GLOBAL_PROTECT(config_error_log)
-
-GLOBAL_VAR(demo_log)
-GLOBAL_PROTECT(demo_log)
-
-GLOBAL_VAR(dynamic_log)
-GLOBAL_PROTECT(dynamic_log)
-
-GLOBAL_VAR(filter_log)
-GLOBAL_PROTECT(filter_log)
-
-#ifdef REFERENCE_DOING_IT_LIVE
-GLOBAL_LIST_EMPTY(harddel_log)
-GLOBAL_PROTECT(harddel_log)
-#endif
-
-GLOBAL_LIST_EMPTY(IClog)
-GLOBAL_PROTECT(IClog)
-
-/// Keeps last 100 signals here in format: "[src] used [REF(src)] @ location [src.loc]: [freq]/[code]"
-GLOBAL_LIST_EMPTY(lastsignalers)
-GLOBAL_PROTECT(lastsignalers)
+GLOBAL_VAR(round_id)
+GLOBAL_PROTECT(round_id)
 
-/// Stores who uploaded laws to which silicon-based lifeform, and what the law was
-GLOBAL_LIST_EMPTY(lawchanges)
-GLOBAL_PROTECT(lawchanges)
+/// The directory in which ALL log files should be stored
 GLOBAL_VAR(log_directory)
 GLOBAL_PROTECT(log_directory)
 
-GLOBAL_VAR(lua_log)
-GLOBAL_PROTECT(lua_log)
+#define DECLARE_LOG_NAMED(log_var_name, log_file_name, start)\
+GLOBAL_VAR(##log_var_name);\
+GLOBAL_PROTECT(##log_var_name);\
+/world/_initialize_log_files(temp_log_override = null){\
```

Comment:
It's a cool trick but as having tried this before, this is going to fill the linter with warnings about overriding the same proc multiple times and will also lead to stack overflow issues if we have too many logs.

This should instead create a datum of some subtype that we can `subtypesof()` later on in one proc.

Reply:
Maybe that's one for JSON logging? IIRC it datumizes logs?

Reply:
I think we're fine for now. This is the exact same trick GLOB does to initialize its vars. We'll stack overflow on those far before we do here.